src/main/java/uk/co/ramp/event/AlertEventProcessor.java

```diff
@@ -115,12 +120,25 @@ public ProcessedEventResult processEvent(AlertEvent event) {
     }
   }
 
-  int timeInStatus(AlertStatus newStatus) {
-
+  int timeInStatus(AlertStatus newStatus, AlertEvent event) {
+    int time = event.time();
     MeanMax progressionData;
     switch (newStatus) {
       case TESTED_POSITIVE:
       case TESTED_NEGATIVE:
+        int conducted = statisticsRecorder.getTestsConducted(time);
+        int capacity =
+            (int) Math.ceil(properties.populationSize() * populationProperties.testCapacity());
+        boolean testsAvailable = conducted < capacity;
+        if (testsAvailable) {
+          statisticsRecorder.recordTestConducted(time);
+          progressionData = ImmutableMeanMax.builder().mean(1).max(1).build();
+        } else {
+          int testTime = findNextTest(time, capacity);
```

Comment:
I wonder if having a queue waiting until a test is available is the right way to handle running out of test capacity - I can imagine a scenario where the backlog builds up, and eventually people are waiting more than 14 days for a test.

Reply:
This is just a first take, Sibylle and Louise seem happy with the simple delay, but we can discuss improvements further.